src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -73,7 +73,13 @@ namespace ts.codefix {
         }
 
         // add the async keyword
-        changes.insertLastModifierBefore(sourceFile, SyntaxKind.AsyncKeyword, functionToConvert);
+        if (functionToConvert.modifiers) {
+            changes.insertModifierAt(sourceFile, functionToConvert.modifiers.end, SyntaxKind.AsyncKeyword, { prefix: " " });
+        }
+        else {
+            const pos = functionToConvert.decorators ? skipTrivia(sourceFile.text, functionToConvert.decorators.end) : functionToConvert.getStart(sourceFile);
+            changes.insertModifierAt(sourceFile, pos, SyntaxKind.AsyncKeyword, { suffix: " " });
+        }

```

Comment:
I think it's a bit easier to read this way:

```suggestion
        const pos = 
            functionToConvert.modifiers ? functionToConvert.modifiers.end
            : functionToConvert.decorators ? skipTrivia(sourceFile.text, functionToConvert.decorators.end) 
            : functionToConvert.getStart(sourceFile);
        const options = functionToConvert.modifiers ? { prefix: " " } : { suffix: " " };
        changes.insertModifierAt(sourceFile, pos, SyntaxKind.AsyncKeyword, options);
```

Also writing it this way makes me wonder: does functionToConvert.modifiers.end also need to be wrapped in a skipTrivia call?

Reply:
@sandersn Thanks for the review 👍. I think it depends. In the current implementation, the `async` keyword will be inserted right after the `public` modifier

```ts
class Foo {
    /* comment */ public async /* commnet */ method() {
        return fetch('a').then(x => x);
    }
}
````

If need to insert `async` after comment and before the function name, I think need to use `skipTrivia` 
```ts
class Foo {
    /* comment */ public /* commnet */ async method() {
        return fetch('a').then(x => x);
    }
}
````
Do I need to change it?

Reply:
No, I think it's better the current way.

Reply:
Ok